cpp/include/cuvs/neighbors/detail/ivf_flat_build.cuh

```diff
@@ -0,0 +1,540 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
```

Comment:
I know you are following the lead of Mickael here, since your changes include his changes. However, I'll reiterate here for purposes of information sharing  that the only headers that should be included in `inclue/` are interfaces and definitions that should be user visible. That means that anything more complex than a trivial enum or POD struct should be getting defined in `src/` and compiled into C++/CUDA source files. 

There's no reason for `include/` to have a `detail` directory at all. Please take a look at the pairwise distances PR for an example of where different files belong.